docs/proposals/resource-aggregation-proxy/README.md

```diff
@@ -0,0 +1,455 @@
+	ResourceSelectors []ResourceSelector `json:"resourceSelectors"`
+}
+
+// ResourceSelector specifies the resources type and its scope.
+type ResourceSelector struct {
+	// APIVersion represents the API version of the target resources.
+	// +required
+	APIVersion string `json:"apiVersion"`
+
+	// Kind represents the plural kind of the target resources.
+	// +required
+	Kind string `json:"kind"`
+
+	// Namespace of the target resource.
+	// Default is empty, which means all namespaces.
+	// +optional
+	Namespace string `json:"namespace"`
+
+	// Selector filters resources by label.
+	// Default is empty, which means no filter.
+	// +optional
+	Selector *metav1.LabelSelector `json:"labelSelector"`
+}
+
+// GlobalResourceStatus defines the observed state of GlobalResource.
+type GlobalResourceStatus struct {
+	// Conditions contain the different condition statuses.
+	// +optional
+	Conditions []metav1.Condition
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GlobalResourceList contains a list of GlobalResource
+type GlobalResourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items holds a list of GlobalResource.
+	Items []GlobalResource
+}
+
+// +k8s:conversion-gen:explicit-from=net/url.Values
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ProxyOptions define a flag for proxy that do not have actual resources.
+type ProxyOptions struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Path is the part of URLs used for the current proxy request.
+	// For example, the whole request URL is
+	// http://localhost/apis/proxy.karmada.io/v1alpha1/globalresources/dummy/proxy/api/v1/nodes
+	// Path is api/v1/nodes
+	// +optional
+	Path string `json:"path,omitempty" protobuf:"bytes,1,opt,name=path"`
+}
+```
+
+### Example
+
+Users create custom `GlobalResource` as below:
+
+```yaml
+apiVersion: proxy.karmada.io/v1alpha1
+kind:  GlobalResource
+metadata:
+  name: proxy-sample
+spec:
+  targetCluster:
+    clusterNames:
+    - member1
+    - member2
+    - member3
+  resourceSelectors:
+    - apiVersion: v1
+      kind: Pod
+    - apiVersion: v1
+      kind: Node
+```
+
+
+
+## How to Merge Result
+
+- Get request: return first result.
+- List request: concat the results, and return to client as a single list.
+- Watch request: return result one by one.
+
+## Resource Version
+
+Kubernetes uses ResourceVersion to uniquely identify every version of a resource. We merge all clusters' ResourceVersion into one JSON struct, `base64.RawURLEncoding` it, and get the final string.
+
+```
+base64.RawURLEncoding({"cluster1":"1234","cluster2":"5678"}) = "eyJjbHVzdGVyMSI6IjEyMzQiLCJjbHVzdGVyMiI6IjU2NzgifQ"
+```
+
+## Example
+
+There are two member clusters: cluster1 and cluster2.
+
+|                 | cluster1         | cluster2         |
+| --------------- | ---------------- | ---------------- |
+| resourceVersion | 1234             | 5678             |
```

Comment:
Is the `1234` the `.metadata.resourceversion` of `Cluster` object?

Reply:
In this table, there are two Pods (`nginx-1` and `nginx-2`), how do you know who is the `latest`?

Reply:
Indeed, we needn't to care which pod is latest. Response from apiserver will tell it, the field `.metadata.resourceversion` of `PodList`.
```
GET /api/v1/namespaces/test/pods
---
200 OK
Content-Type: application/json

{
  "kind": "PodList",
  "apiVersion": "v1",
  "metadata": {"resourceVersion":"1234"},
  "items": [...]
}
```

Reply:
Oh, good to know. I didn't notice the `List` also has the `resourceVersion`. Thanks for the clarification. 
Do you think we should add it to the proposal? I think it would get easy for other people to understand.

Reply:
updated